### PR TITLE
Ekaterina pogodina assignment2

### PR DESCRIPTION
**Pull request by EkaterinaPogodina**

### CODE REVIEW

File: EkaterinaPogodina/assignment-1/README.md [hunk 1]
```diff
@@ -1,12 +1,9 @@
 # Assignment 1
```
[the remark this answers is not in the document]
Comment (EkaterinaPogodina) [reply EkaterinaPogodina/assignment-1/README.md]: Sorry, I wasn't careful in this question.

File: EkaterinaPogodina/assignment-2/binary_tree.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            ancestor (Node): parent of the node
+            add_left (bool): flag determines whether add child to left or to right
+        """
+        self.left = None
+        self.right = None
+        self.parent = None
+        self.key = key
+        self.depth = 0
+
+        if ancestor is not None:
```
[the remark this answers is not in the document]
Comment (EkaterinaPogodina) [reply EkaterinaPogodina/assignment-2/binary_tree.py]: Ok, fixed

File: EkaterinaPogodina/assignment-2/binary_tree.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            self.depth = ancestor.depth + 1
+            self.parent = ancestor
+
+            if ancestor.left is None and add_left:
+                ancestor.left = self
+
+            else:
+                ancestor.right = self
```
[the remark this answers is not in the document]
Comment (EkaterinaPogodina) [reply EkaterinaPogodina/assignment-2/binary_tree.py]: I thought that's enough to write "I assume, that ancestor doesn't have either left child or right."

File: EkaterinaPogodina/assignment-2/example.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+"""Example of usage binary_tree.py"""
+
+from binary_tree import Node, BinaryTree
+from binary_tree import find_node, find_ancestors, find_common_ancestor
+
+root = Node(7)
+tree = BinaryTree(root)
+
+tree.add_node(3, 7, True)
```
[the remark this answers is not in the document]
Comment (EkaterinaPogodina) [reply EkaterinaPogodina/assignment-2/example.py]: yes, fixed

File: EkaterinaPogodina/assignment-2/binary_tree_test.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        tree.add_node(4, 7)
+        tree.add_node(2, 3)
+        tree.add_node(5, 3)
+        tree.add_node(8, 4)
+        tree.add_node(1, 2)
+        tree.add_node(6, 2)
+
+        self.assertEqual(binary_tree.find_ancestors(tree, 6), [6, 2, 3, 7])
+
+    def testFindCommonAncestor(self):
```
[the remark this answers is not in the document]
Comment (EkaterinaPogodina) [reply EkaterinaPogodina/assignment-2/binary_tree_test.py]: I wasn't careful...

File: EkaterinaPogodina/assignment-2/binary_tree.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+        """
+        Creating node of the tree
+
+        Args:
+            key (Any): key of the node to be created
+            parent (Any): key of the parent
+            add_left (bool): flag determines whether add node to left or to right
+        """
+        parent_node = find_node(self.root, parent)
+        node = Node(key, parent_node, add_left)
```
[the remark this answers is not in the document]
Comment (EkaterinaPogodina) [reply EkaterinaPogodina/assignment-2/binary_tree.py]: fixed

File: EkaterinaPogodina/assignment-2/binary_tree.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    """
+    def _find_parent(node):
+        """
+        Appending ancestor of the node to all_ancestors list
+        """
+        if node.parent is None:
+            return
+
+        all_ancestors.append(node.parent.key)
+        _find_parent(node.parent)
```
[the remark this answers is not in the document]
Comment (EkaterinaPogodina) [reply EkaterinaPogodina/assignment-2/binary_tree.py]: ok

File: EkaterinaPogodina/assignment-2/binary_tree_test.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        tree = binary_tree.BinaryTree(root)
+
+        node = binary_tree.find_node(tree.root, 10)
+        self.assertIsNone(node)
+
+    def testFindNode(self):
+        root = binary_tree.Node(7)
+        tree = binary_tree.BinaryTree(root)
+
+        tree.add_node(3, 7, True)
```
[the remark this answers is not in the document]
Comment (EkaterinaPogodina) [reply EkaterinaPogodina/assignment-2/binary_tree_test.py]: Ok. Maybe me representation is not the best, but I tried.

[the remark this answers is not in the document]
Comment (simonhollingshead) [reply EkaterinaPogodina/assignment-2/binary_tree_test.py]: I apologise if I wasn't clear with my comment.  I never intended for you to implement a repr for this, I just meant that at the top of the test case you could include a comment explaining what the test data structure looked like.  It's awesome code, though, so thank you for it, but I do want to make clear that I did not intend for you to need to do all that extra work!

**Pull request merged**